### DISCUSSION TITLE
Request for some customization in Dropdown (Editable) of PrimeNG-V9

### DISCUSSION THREAD

**Discussion by Pawan-Cavisson** [primefaces/community#480] [PrimeNG] [answered]
Hi,
I'm using PrimeNG-V9 and looking to implement a dropdown with an input box, similar to the existing "**Editable**" in the Dropdown, PrimeNG-V9. So, I want some customization in "**Editable**" Dropdown.

**Dropdown** PrimeNG-V9 : https://www.primefaces.org/primeng-v9-lts/#/dropdown

### **Required Customization**
**1. Open Dropdown Menu**
Currently, when the user clicks on the dropdown icon, the dropdown menu opens. I want to modify this behavior so that as soon as the user clicks on the input box, the dropdown menu should open.

**2. Item of dropdown menu**
Currently, Each item in the dropdown menu contains only one text value. Now the requirement is Each Item will contain 2 divs(div refers to div tag in HTML) Each div will have some Text and each can have its own CSS Style.

**3. Support of filter functionality**
As the filter functionality is working in **"Content with filters"** in dropdown, PrimeNG-V9. Similarly, in our case when user will type something in input box the filter functionality should apply on the dropdown menu.

### **Resultant Dropdown**

1. There will be an input box. As soon as user will click on input box the dropdown menu should open
2. Each Item will contain 2 divs(div refers to div tag in HTML) Each div will have some Text and each can have its own CSS Style.
3. When user will type something in input box the filter functionality should apply on the dropdown menu.
4. User can use Up/Down Arrow key to navigate the dropdown menu.
5. When user select any Item of dropdown using Up/Down Arrow key and then hits Enter button the selected item should become the selected value.
6. When user types something in inputbox which is not listed in dropdown menu and then hits Enter button. The entered text should become the selected value.

I have tried to achieve this using `<ul>` and `<li>` to achieve the same.
[Required Dropdown Sample.docx](https://github.com/primefaces/community/files/13375598/Required.Dropdown.Sample.docx)

**Comment by cetincakiroglu** [comment, marked as answer]
1-) You cannot do this on v9; you can access the component view and manipulate the click logic.
2-) You can do it by using the itemTemplating option explained in the v9 dropdown docs, please read the docs carefully.
3-) Dropdown doesn't have a feature like this.
4-) Keyboard support was updated in v16 as in the accessibility update. If you need to use more recent features, please update your version. v9 will not receive an update because it's outdated.
